dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -907,4 +919,66 @@ public Group createDefaultReadGroup(Context context, Collection collection, Stri
         return role;
     }
 
+    @Override
+    public List<Collection> findAuthorizedCollectionsInSOLR(String q, Context context, Community community,
+        int offset, int limit) throws SQLException, SearchServiceException {
+
+        List<Collection> collections = new ArrayList<Collection>();
+        StringBuilder query = new StringBuilder();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
+        discoverQuery.setStart(offset);
+        discoverQuery.setMaxResults(limit);
+        DiscoverResult resp = resultSolrQuery(context, query, discoverQuery,community, q);
+        for (IndexableObject solrCollections : resp.getIndexableObjects()) {
+            Collection c = ((IndexableCollection) solrCollections).getIndexedObject();
+            collections.add(c);
+        }
+        return collections;
+    }
+
+    @Override
+    public int countAuthorizedCollectionsInSOLR(String q, Context context, Community community)
+        throws SQLException, SearchServiceException {
+
+        StringBuilder query = new StringBuilder();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
+        DiscoverResult resp = resultSolrQuery(context, query, discoverQuery,community,q);
+        return (int)resp.getTotalSearchResults();
+    }
+
+    private DiscoverResult resultSolrQuery(Context context, StringBuilder query, DiscoverQuery discoverQuery,
+                                   Community community, String q) throws SQLException, SearchServiceException {
+
+        EPerson currentUser = context.getCurrentUser();
+        if (!authorizeService.isAdmin(context)) {
+            Group anonymousGroup = groupService.findByName(context, Group.ANONYMOUS);
```

Comment:
Can we leave out the call to the anonymousGroup group here ? Anonymous will never be granted submit rights (as you always need a logged in user), so this database call is not needed.